### ISSUE TITLE
Logs do not show resource names

### ISSUE THREAD

**Issue by srm09** [kubernetes-sigs/cluster-api-provider-vsphere#1580]
/kind bug

**What steps did you take and what happened:**
Checking the log output for the capv manager pod, we see these entries
```log
I0719 18:34:52.688028       1 vspheremachine_controller.go:159] capv-controller-manager/vspheremachine-controller "msg"="Waiting for Machine Controller to set OwnerRef on VSphereMachine"
I0719 18:34:52.688471       1 vspheremachine_controller.go:159] capv-controller-manager/vspheremachine-controller "msg"="Waiting for Machine Controller to set OwnerRef on VSphereMachine"
I0719 18:34:52.692036       1 vspheremachine_controller.go:159] capv-controller-manager/vspheremachine-controller "msg"="Waiting for Machine Controller to set OwnerRef on VSphereMachine"
```

**What did you expect to happen:**
We expect the name of the VsphereMachine to be appended for easier debugging purposes.
```log
I0719 18:34:52.692036       1 vspheremachine_controller.go:159] capv-controller-manager/vspheremachine-controller/<foo-machine> "msg"="Waiting for Machine Controller to set OwnerRef on VSphereMachine"
```

**Anything else you would like to add:**
n/a

**Environment:**

- Cluster-api-provider-vsphere version: v1.3.1
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by ameddin73** [comment]
I will try to work on this. Be patient - it's my first issue :)

**Comment by ameddin73** [comment]
I couldn't get it to build. Looks like I don't have enough experience with VSphere :/
/unassign @ameddin73

**Comment by AectannArd** [comment]
@srm09 
can you confirm that the issue is described properly?

1. going through [the code](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/vspheremachine_controller.go#L197) this is the only place I can find which seems to be textually close to the logs you provided (but it is doesn't match exactly)
2. the exact match was last met in [release v1.0.1](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/v1.0.1/controllers/vspheremachine_controller.go#L159) due to the fact that the direct construct `r.Logger` is used
3. since then the message has been rephrased to match the recent text (in [v1.1.0-alpha](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/v1.1.0-alpha/controllers/vspheremachine_controller.go#L195)) and bug-fixed to to use correct logger by you (in [v1.3.0](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/v1.3.0/controllers/vspheremachine_controller.go#L184))
4. though I did not reproduce the issue with missing OwnerRef in tests going through the logs that I was able to acquire gives us the following message ```
I1001 18:34:39.335036   27303 vspheremachine_controller.go:299] "capv-controller-manager/vspheremachine-controller/vsphere-machine-reconciler-vgf9n/vsphere-machine-1: Cluster infrastructure is not ready yet"
``` which comes from [context logger](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/vspheremachine_controller.go#L299) which in order is set [while building base context](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/vspheremachine_controller.go#L216) from the same logger that is [used to print the OwnerRef](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/vspheremachine_controller.go#L197) message.


NB. The only place in this controller that is using non-contexted logger seem to be [fetchCAPICluster(...) method](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/vspheremachine_controller.go#L346-L358) but it's definitely not described in this task

**Comment by AectannArd** [comment]
@srm09
could you have a look please?

**Comment by AectannArd** [comment]
Seems that the issue shouldn't have been migrated to new release scope. It should have been closed.

**Comment by srm09** [comment]
@AectannArd Good catch, the particular instance of the logger mentioned in the issue was fixed by the PR [#1544](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1544)
/close
